src/Databases/SQLite/DatabaseSQLite.cpp

```diff
@@ -30,15 +34,19 @@ DatabaseSQLite::DatabaseSQLite(
     , database_engine_define(database_engine_define_->clone())
     , log(&Poco::Logger::get("DatabaseSQLite"))
 {
+    std::error_code err;
+    String canonical_path = fs::canonical(database_path_, err);
+    /// The path existence is also checked.
+    if (err)
+        throw Exception(ErrorCodes::PATH_ACCESS_DENIED, "SQLite database path '{}' is invalid. Error: {}", database_path_, err.message());
+
     sqlite3 * tmp_sqlite_db = nullptr;
-    int status = sqlite3_open(database_path_.c_str(), &tmp_sqlite_db);
+    int status = sqlite3_open(canonical_path.c_str(), &tmp_sqlite_db);
```

Comment:
Add support function that returns std::shared_ptr<sqlite3>, that will check that status is ok and path is ok.